apps/explorer/src/pages/transaction-result/programmable-transaction-view/Inputs.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiCallArg } from '@mysten/sui.js';
+
+import { AddressLink, ObjectLink } from '~/ui/InternalLink';
+import { TableHeader } from '~/ui/TableHeader';
+
+interface Props {
+    inputs: SuiCallArg[];
+}
+
+export function Inputs({ inputs }: Props) {
+    if (!inputs?.length) {
+        return null;
+    }
+
+    return (
+        <>
+            <TableHeader>Inputs</TableHeader>
+            <ul className="flex flex-col gap-y-3">
+                {inputs.map((input, index) => {
+                    if (typeof input !== 'object') {
+                        return (
+                            <li key={index}>
+                                <AddressLink
+                                    noTruncate
+                                    address={String(input)}
+                                />
+                            </li>
+                        );
+                    }
+
+                    if ('valueType' in input && 'value' in input) {
+                        if (input.valueType === 'address') {
+                            return (
+                                <li key={index}>
+                                    <AddressLink
+                                        noTruncate
+                                        address={String(input.value)}
+                                    />
+                                </li>
+                            );
+                        }
+
+                        if (input.type === 'pure') {
```

Comment:
This should be implied by there being an `input.value`?

Reply:
This is done on `if ('valueType' in input && 'value' in input)`, this is nested inside that in the event `valueType` is not `address`

Reply:
I know but I'm saying I don't think you need this check at all, given there being an input implies the type is pure.